streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -1121,14 +1114,14 @@ public void onAssignment(final Assignment assignment, final ConsumerGroupMetadat
         final Map<TopicPartition, TaskId> partitionsToTaskId = new HashMap<>();
 
         switch (receivedAssignmentMetadataVersion) {
-            case VERSION_ONE:
+            case 1:
```

Comment:
These constants were nice in spirit, but they are a maintenence liability:
* There's no enforcable restriction that they must be used... Code review is manifestly not sufficient, since there have already been many PRs merged that use the literals instead of the constants.
* The constants are almost vapid, in that the name is the same as the number they refer to. The extra information they contain is that the number represents an "assignment protocol version", but looking at the code, context is universally sufficient to determine that fact. For example, in this usage, we're switching on receivedAssignmentMetadataVersion, so obviously the numbers in the case statements are version numbers. All the usages of these constants have the same property.

I could go on, but these two points above should be sufficient just to inline the constants and be done with it.